batcher/test/common/helpers.mligo

```diff
@@ -0,0 +1,260 @@
+  (level: Breath.Logger.level)
+  (tzbtc_trader: Breath.Context.actor)
+  (usdt_trader: Breath.Context.actor)
+  (eurl_trader: Breath.Context.actor)
+  (batch: Batcher.batch)
+  (pair: string) =
+  let oracle = originate_oracle level in
+  let additional_oracle = originate_oracle level in
+  let tzbtc = originate_tzbtc tzbtc_trader level in
+  let usdt = originate_usdt usdt_trader level in
+  let eurl = originate_eurl eurl_trader level in
+  let initial_storage = TestStorage.initial_storage oracle.originated_address tzbtc.originated_address usdt.originated_address eurl.originated_address in
+  let batch_set = initial_storage.batch_set in
+  let cbi = Map.add pair 1n batch_set.current_batch_indices in
+  let batches = Big_map.add 1n batch batch_set.batches in
+  let batch_set = { batch_set with 
+    current_batch_indices = cbi;
+    batches = batches
+  } in
+  let initial_storage = { initial_storage with batch_set = batch_set; } in
+  let batcher = TestUtils.originate initial_storage level in
+  {
+   batcher = batcher;
+   oracle = oracle;
+   additional_oracle = additional_oracle;
+   tzbtc = tzbtc;
+   usdt = usdt;
+   eurl = eurl;
+  }
+
+let test_context
+    (level: Breath.Logger.level) = 
+      let (_, (btc_trader, usdt_trader, eurl_trader)) = Breath.Context.init_default () in
+      let fee_recipient_address = usdt_trader.address in 
+      let contracts = originate_with_admin_and_fee_recipient level btc_trader usdt_trader eurl_trader eurl_trader fee_recipient_address in
+      {
+        btc_trader = btc_trader;
+        usdt_trader = usdt_trader;
+        eurl_trader = eurl_trader;
+        admin = eurl_trader;
+        non_admin = btc_trader;
+        fee_recipient = fee_recipient_address;
+        contracts = contracts;
+      }
+
+let test_context_with_batch
+    (pair: string)
+    (batch: Batcher.batch)
+    (level: Breath.Logger.level) = 
+      let (_, (btc_trader, usdt_trader, eurl_trader)) = Breath.Context.init_default () in
+      let fee_recipient_address = usdt_trader.address in 
+      let contracts = originate_with_batch_for_clearing level btc_trader usdt_trader eurl_trader batch pair in
+      {
+        btc_trader = btc_trader;
+        usdt_trader = usdt_trader;
+        eurl_trader = eurl_trader;
+        admin = eurl_trader;
+        non_admin = btc_trader;
+        fee_recipient = fee_recipient_address;
+        contracts = contracts;
+      }
+
+let create_order
+  (from: string)
+  (to: string)
+  (amount: nat)
+  (side: Batcher.side)
+  (tolerance: Batcher.tolerance)
+  (valid_tokens: Batcher.valid_tokens) : Batcher.external_swap_order =
+  let fromToken = Map.find from valid_tokens in
+  let toToken = Map.find to valid_tokens in
+  let nside = TestUtils.side_to_nat side in
+  let swap = {
+     from = {
+       token = fromToken;
+       amount = amount;
+     };
+     to = toToken;
+  } in
+  let ntol = TestUtils.tolerance_to_nat tolerance in
+  {
+    swap = swap;
+    created_at = Tezos.get_now ();
```

Comment:
I'm not sure you should use `get_now` when you create orders, as it can be sensitive to the execution context of the test. Maybe just forge timestamps and pass them around as argument?

Reply:
None of the tests really rely on it so it is just an easy timestamp generator.